requirements.txt

```diff
@@ -21,6 +21,7 @@ flake8-docstrings>=1.4.0
 flake8-quotes>=2.1.0
 flake8-tuple>=0.4.0
 isort==5.9.3
+jax==0.3.15  # Needed by TF 2.12. Versions 0.4.* use self-documenting f-strings, which are incompatible with Python 3.7. Later patches of 0.3.* cause other mypy errors.
```

Comment:
Is this a good time to unpin click?

Reply:
non-blocking: Could this also be `jax<=0.3.15`?

Reply:
Nope. Tensorflow 2.12 requires `jax>=0.3.15`. I pinned it at `0.3.15` because later versions caused weird errors with mypy.

Reply:
Thanks. I hadn't understood from the existing comment.

Reply:
Could you please take a look at the "click" pin and decide what you want to do with it?

Reply:
https://hpe-aiatscale.atlassian.net/browse/MLG-445